### ISSUE TITLE
Light refactor of Helm chart install/upgrade component

### ISSUE THREAD

**Issue by catherineluse** [rancher/dashboard#6069]
This issue is a follow-up to https://github.com/rancher/dashboard/pull/6068. While I was documenting the logic for how Rancher handles Helm chart app values, I saw some opportunities to refactor [https://github.com/rancher/dashboard/blob/release-2.6.5/pages/c/_cluster/apps/charts/install.vue](https://github.com/rancher/dashboard/blob/release-2.6.5/pages/c/_cluster/apps/charts/install.vue).

Here are the three changes I would make:

- Avoid having two really long functions - `fetch` and `actionInput` - where all the logic is happening. It would be easier to read and understand if, within those functions, we had some logic broken out into smaller functions with names like `setUpValuesForInstall` or `setUpValuesForUpgrade`.
- Avoid removing global values just to re-add them again. Currently we have two functions - removeGlobalValuesFrom and addGlobalValuesTo. Each does the opposite. removeGlobalValuesFrom removes each value from global values if it is equal to what is the current and correct value. Then addGlobalValuesTo calls `setIfNotSet` to re-add each value again. It would be better to use the lodash `merge` function (which is already used elsewhere in the file) to handle all the global values in `actionInput` at the very end of the workflow. This would make it harder to introduce some error when we change or add a global value, which we had to do for project monitoring.
- Convert it to typescript. While documenting the component I had to write out which values were strings, and what keys were in certain returned objects. It would be cleaner if that was in Typescript.

**Comment by codyrancher** [comment]
@momesgin given all the changes and re-design work you've been doing do you think we can close this issue?

**Comment by momesgin** [comment]
@codyrancher I don't think that we can close this yet, I'll assign this to myself to work on it later

**Comment by codyrancher** [comment]
Alright, I added some of the appropriate labels and project.

**Comment by momesgin** [comment]
Changing the milestone since the install page is being redesigned. It makes more sense to handle this refactoring alongside the implementation of the new design.
